quill-jdbc-zio/src/test/scala/io/getquill/misc/OnDataSourceSpec.scala

```diff
@@ -56,6 +57,15 @@ class OnDataSourceSpec extends PeopleZioProxySpec {
 
       people mustEqual peopleEntries.filter(p => p.name == "Alex")
     }
+    "should keep existing errors" in {
+      // This is how you import the encoders/decoders of `underlying` context without importing things that will conflict
+      // i.e. the quote and run methods
+      import testContext.underlying.{prepare => _, run => _, _}
+      import java.sql.SQLException
+
+      val zioThatCanFail: ZIO[Connection, String, Nothing] = ZIO.service[Connection] *> ZIO.fail("Custom Error")
+      "val result: ZIO[DataSource, String | SQLException, List[Person]] = zioThatCanFail.onDataSource" should compile
```

Comment:
I wrote the tests in this style (`something must compile`) because the only thing that changed is the types. Let me know if you have any suggestions!